text/0249-v2-experiments.md

```diff
@@ -0,0 +1,127 @@
+
+The contributor guide can be copied from the main repository, since the toolchain for building the preview repository
+will be basically the same.
+
+# Design Summary
+
+- Upon release of v2.0 to GA, all experimental modules will be automatically marked as stable (in fact, the metadata in
+  the code regarding stability can be removed, since everything will be stable). This means that the opportunity for the
+  team and for the community to influence the shape of existing experimental APIs is between now and then. Once we go
+  GA, we have to live with our design decisions.
+- Any APIs that cannot be graduated with confidence in time for the v2 release will be deprecated in 1.x, removed from
+  the v2 branch, and moved to the preview repository. This option should be minimized, since some customers may have
+  deployed production stacks that depend on these APIs.
+- Create a new repository called aws-cdk-previews that will be the new home for experimental development and developer
+  preview releases. The packages in this repository will be vended in the same way that 1.x packages are vended today,
+  in a variety of languages, with metadata that draws a line between new experiments and more mature developer preview
+  packages. This repo will take a dependency on the core CDK package just like any other independent library. This
+  avoids the complications associated with experimenting on a fork or a branch of the main repository. The repository
+  will be clearly marked as experimental.
+- When a package in the preview repository has reached maturity and is ready for GA, it will be copied into the main
+  repo and made a part of CDK 2. The module in the preview repository will then be marked as deprecated with clear
+  documentation to point users to the new stable version in the main repository. The lifecycle of experimental modules
+  in the preview repository will be exactly the same as it is currently in 1.x, with the same metadata markers in the
+  code.
+- The contributor guide for the CDK will be modified so that users know where to introduce new experimental code.
+- If it becomes absolutely necessary to add a new feature to v2 that would be a breaking change for users, the feature
+  will be hidden behind a feature flag. The code path will not be accessible without a specific opt-in by the user.
+
+# Drawbacks
+
+- In several places in the current 1.x CDK branch, stable modules refer to or contain experimental code. This will no
+  longer be possible after the release of v2.
+- When we graduate a module, users who have participated in the Developer Preview will have to update their imports to
+  make use of the stable version in the main repository.
+- For developers who are working on experimental code that has complex dependencies with the core and with other
+  modules, development will be more complicated.
+- In some cases, experimentation will not be possible. New features introduced to the main repository will have to be
```

Comment:
You're throwing this drawback out pretty casually as a statement of fact.

Convince me why it is okay to lose development speed.

Reply:
I think the loss in speed on the stable repo will be compensated by much faster development on the preview repository. I also anticipate more contributors there, since it will be much easier to get started.